### ISSUE TITLE
Extend MockRetrofit to inject a CoroutineDispatcher

### ISSUE THREAD

**Issue by steve-the-edwards** [square/retrofit#3360]
Feature Request: Extend MockRetrofit to allow users to specify a `CoroutineDispatcher` instead of just an `ExecutorService` for background work.

This will allow for synchronous tests using MockRetrofit with a coroutine based API.

**Comment by JakeWharton** [comment]
Can you pass a synchronous executor to achieve this?

**Comment by steve-the-edwards** [comment]
Yes, you can.  It may be nice to be able to have an Executor backed by a TestCoroutineDispatcher() though.  Again, something you can do yourself when building it - I just thought it might be a nice addition to the API.
